web-app/HeimScripts/heatmap/summary.R

```diff
@@ -76,17 +117,29 @@ check_input <- function(datasets, phase_info, projection)
                "\ne.g. n0_s1, n0_s2, n1_s1, n1_s2. "))
   }
   
-  
-  if(is.na(phase_info))
+  #all labels of the data.frames should be unique
+  dataset_names <- names(datasets)
+  if(any(duplicated(dataset_names)))
   {
-    stop("Supply phase parameter to function \'main()\'. Expected input: \'fetch\' or \'preprocess\'")
+    stop(paste("Not all data.frame labels are unique; one or more labels in the \'loaded_variables\' or \'preprocessed\' variable are duplicated"))
   }
   
-  if(phase_info != "fetch" & phase_info != "preprocess" & !is.na(phase_info))
+  #check if data.frame "preprocessed" has the right column names
+  # expected format of column names for sample columns is <sample_name>_n<numerical node id>_s<numerical subset id>, eg. sample234_n1_s1
+  if(dataset_names[1] == "preprocessed") #for preprocessed data there is only 1 data.frame
   {
-    stop("Incorrect value for phase parameter - expected input: either \'fetch\' or \'preprocess\'")
+    column_names <- colnames(datasets$preprocessed)
+    expected_format_names <- ".+_n[[:digit:]]+_s[[:digit:]]+$"
```

Comment:
yes, I know. The code just expects numerical identifier for subset, without the additional constraint that it needs to be 1 or 2, and it should work with any numerical identifier (that is not too long. I didn't test anything else than 1 or 2 however). I made it a bit more flexible on purpose, so it can handle more subsets or different numerical identifiers (for example if later the subsets are identified with some numeric value -- e.g. if subset settings are saved and later retrieved again or something)